src/components/FileTree/index.spec.tsx

```diff
@@ -389,5 +389,19 @@ describe(__filename, () => {
 
       expect(node.props.onSelect).toEqual(onSelect);
     });
+
+    it('passes the version ID to FileTreeNode', () => {
+      const version = getVersion({ ...fakeVersion, id: 777 });
+
+      const root = render({ version });
+
+      const node = (root.instance() as FileTree).renderNode(
+        // It does not really matter which props are given here.
+        // eslint-disable-next-line @typescript-eslint/no-object-literal-type-assertion
+        {} as TreefoldRenderProps<TreeNode>,
+      );
+
+      expect(node.props.versionId).toEqual(version.id);
```

Comment:
Hmm, I've never seen tests work directly with JSX objects like this. Is this a stable API? I would feel better about doing shallow rendering which we know is a reliable API:

```diff
diff --git a/src/components/FileTree/index.spec.tsx b/src/components/FileTree/index.spec.tsx
index faae880..44e9cb4 100644
--- a/src/components/FileTree/index.spec.tsx
+++ b/src/components/FileTree/index.spec.tsx
@@ -13,6 +13,7 @@ import {
 } from '../../reducers/versions';
 import { fakeVersion, fakeVersionEntry } from '../../test-helpers';
 import { getLocalizedString } from '../../utils';
+import FileTreeNode from '../FileTreeNode';
 
 import FileTree, { buildFileTree, TreeNode, DirectoryNode } from '.';
 
@@ -401,7 +402,10 @@ describe(__filename, () => {
         {} as TreefoldRenderProps<TreeNode>,
       );
 
-      expect(node.props.versionId).toEqual(version.id);
+      expect(shallow(<div>{node}</div>).find(FileTreeNode)).toHaveProp(
+        'versionId',
+        version.id,
+      );
     });
   });
 });
```

Reply:
That works for me, although we get a react component in `node`, and `props` is definitely part of the public/stable API.